Modules/RatMama.py

```diff
@@ -0,0 +1,178 @@
+                                         # Wrap the entirety of rest of the pattern in a group to make it easier to echo the entire thing
+    (?P<all>
+    (?P<cmdr>[^\s].*?)                   # Match CDMR name.
+    \s+-\s+                              #  -
+    System:\s*(?P<system>.*?)            # Match system name
+    (?:\s[sS][yY][sS][tT][eE][mM]|)      # Strip " system" from end, if present (case insensitive)
+    \s+-\s+                              #  -
+    Platform:\s*(?P<platform>\w+)        # Match platform (currently can't contain spaces)
+    \s+-\s+                              #  -
+    O2:\s*(?P<o2>.+?)                    # Match oxygen status
+    \s+-\s+                              #  -
+    Language:\s*
+    (?P<full_language>                   # Match full language text (for regenerating the line)
+    (?P<language>.+?)\s*                 # Match language name. (currently unused)
+    \(                                   # The "(" of "(en-US)"
+    (?P<language_code>.+?)               # "en"
+    (?:                                  # Optional group
+        -(?P<language_country>.+)        # "-", "US" (currently unused)
+    )?                                   # Actually make the group optional.
+    \)                                   # The ")" of "(en-US)"
+    )                                    # End of full language text
+    (?:                                  # Possibly match IRC nickname
+    \s+-\s+                              #  -
+    IRC\s+Nickname:\s*(?P<nick>[^\s]+)   # IRC nickname
+    )?                                   # ... emphasis on "Possibly"
+    )                                    # End of the main capture group
+    \s*                                  # Handle any possible trailing whitespace
+    $                                    # End of pattern
+""")
+
+
+@rule(r"^\s*Incoming Client:", case_sensitive=False, full_message=True, prefixless=True,
+      pass_match=False)
+async def handle_ratmama_announcement(ctx: Context):
+    if ctx.user.nickname.casefold() not in ('ratmama[bot]',):
+        return
+
+    message: str = ctx.words_eol[0]
+    result = re.fullmatch(ratmama_regex, message)
+    client_name: str = result.group("cmdr")
+    system_name: str = result.group("system")
+    platform_name: str = result.group("platform")
+    o2_status: bool = result.group("o2") == "OK"
+    lang_code: str = result.group("language_code")
+    nickname: Optional[str] = result.group("nick")
+
+    exist_rescue: Rescue = board.find_by_name(client_name)
+
+    if exist_rescue:
+        # we got a case already!
+        await ctx.reply(f"{client_name} has reconnected! #{exist_rescue.board_index}")
+        # now let's make it more visible if stuff changed
+        diff_response: str = ""
+        if system_name.casefold() != exist_rescue.system.casefold():
+            diff_response += f"System changed! Was {exist_rescue.system}"
```

Comment:
Needs whitespace (also punctuation might serve to provide an atmosphere of professionalism to the client).